app/main/views/services.py

```diff
@@ -59,14 +70,17 @@ def edit_service(service_id):
     )
 
 
-@main.route('/services/<string:service_id>/remove', methods=['POST'])
+@main.route("/frameworks/<string:framework_slug>/services/<string:service_id>/remove", methods=['POST'])
 @login_required
-def remove_service(service_id):
+def remove_service(framework_slug, service_id):
     service = data_api_client.get_service(service_id).get('services')
```

Comment:
So this is where half our api functions pass through a 404 and half of them swallow it and return `None`.

Anyway, this bit ain't my code and my way of properly "fixing" it would involve doing all the object fetching using decorators.